src/client/types.ts

```diff
@@ -159,7 +160,7 @@ type PathToChain<
     }
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type Client<T> = T extends Hono<any, infer S, any>
+export type Client<T> = T extends HonoBase<any, infer S, any>
```

Comment:
Weird again.

The reason why I added `ExcludeEmptyObject` is that schema type with empty object type broke RPC type.
For example, `T` in code below is `never` without `ExcludeEmptyObject`.
```typescript
const app = new Hono().route('/', new Hono().get('/foo', c => c.json({ ok: true })))
type T = Client<typeof app>
```

But, replacing `Hono` with `HonoBase` in `Client` fixed the problem

Reply:
🤔 
```typescript
const app = new Hono()
  .route('/', new Hono().get('/foo', c => c.json({ ok: true })))
type Extract1<T> = T extends Hono<any, infer S, any> ? S : never
type Extract2<T> = T extends HonoBase<any, infer S, any> ? S : never
type R1 = Extract1<typeof app> // {}
type R2 = Simplify<Extract2<typeof app>> // {} | { '/foo': ... }
```

Reply:
@m-shaka 

Is that behavior related to the following weird stuff?:

![CleanShot 2024-10-14 at 18 35 10@2x](https://github.com/user-attachments/assets/654e8f01-03ea-4a47-b5ff-101417bbd94e)

![CleanShot 2024-10-14 at 18 36 12@2x](https://github.com/user-attachments/assets/913ac4c1-61cb-4c34-bc10-0ad10f6db3f8)

Reply:
I think that's a matter of display.
The return type of `route` is `HonoBase` and it's correct, but in `src/hono-base.ts` its name is `Hono`. If you rename class `Hono` with `HonoBase` in `src/hono-base.ts`, you see always `HonoBase` in the widget. Presumably, the compiler is kindly renaming it to distinguish `Hono` and `HonoBase`.

<img width="753" alt="image" src="https://github.com/user-attachments/assets/ef329287-8eba-4326-aa05-c8b9739d5d04">

Reply:
@m-shaka 

Ah, understand. Thanks!